cylc/flow/run_modes/simulation.py

```diff
@@ -162,7 +162,7 @@ def __init__(
             db_info = db_mgr.pri_dao.select_task_job(
                 itask.tokens['cycle'],
                 itask.tokens['task'],
-                itask.tokens['job'],
+                itask.submit_num,
```

Comment:
Unrelated fix spotted by MyPy as the result of type changes on this branch:

* `TaskProxy.tokens['job']` is always `None`.
* This caused the DB check to retrieve the latest job, which is probably what we want.
* Made this explicit by passing through the actual job number.